### PR TITLE
Bug:  Apple doesn't support passing both a currencyCode and numberStyle

### LINKED ISSUE #38: passbook generates both numberStyle and currencyCode on the balance field
The passbook library is generating passes that are not loadable by IOS due to the fact that it defaults 'numberStyle' on a the balance header.  This is reproduced by explicitly setting the currencyCode as shown in the unit tests in the attached PR.

I submitted pull #37 as a way to reproduce the problem with a possible fix.  Thanks for your consideration.

John

### PR DESCRIPTION
**Pull request by johnksterling**
We are running into a problem where we set the currencyCode on the balance field of a pass, but never set the numberStyle - yet the passbook python library re-injects a default numberStyle.  Apple, however, doesn't support having both a numberStyle and a currency field.  

There are a few ways to resolve this, but I chose to just over-ride the json generation on CurrencyField to only strip numberStyle if currencyCode is present (which looks to be almost always, unless someone explicitly deletes one of them).

Note that the proposed solution will leave 'empty' the json as-is if one of them is an empty string.  I am not sure if this is the desired behavior since we don't set either to empty string - bit it would be an easy change to check json_dictionary.get('currencyCode')  and json_dictionary.get('currencyCode').strip() if that was preferred.

I have tested this with a couple of version of the ios and it seems to work well.  Anyone else running into this and interested in fixing it?

Closes #38

### CODE REVIEW

File: .cache/v/cache/lastfailed [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (mbaechtold) [inline .cache/v/cache/lastfailed]: Could you please add a line to the `.gitignore` file, so this file does not get commit:

```
# Unit test / coverage reports
.coverage
.tox
.cache   # <--- Add this line.
```

File: passbook/test/test_passbook.py [hunk 2]
```diff
@@ -98,6 +98,40 @@ def test_auxiliary_fields():
...
+        float(22.00),
+        'test label',
+        None
+        )
+    del balance_field.currencyCode
+    balance_field.changeMessage = 'Balance changed to %@.'
+    passfile.passInformation.headerFields.append(balance_field)
+    passfile.passInformation.addAuxiliaryField('aux1', 'VIP Store Card', 'Famous Inc.')
+    pass_json = passfile.json_dict()
+    print(pass_json)
```
Comment (mbaechtold) [inline passbook/test/test_passbook.py]: I think this line should be removed?

File: .cache/v/cache/lastfailed (hunk 1 above)
Comment (johnksterling) [inline .cache/v/cache/lastfailed]: Done

Comment (mbaechtold) [inline .cache/v/cache/lastfailed]: Thanks, but unfortunately I forgot to mention that you must delete this file too.

**Comment by driskell** [comment]
@johnksterling @mbaechtold Hi guys. Did something need to happen to merge this? I can see it didn't reach the master. Thanks!

**Comment by driskell** [comment]
Would just changing `CurrencyField(NumberField)` to `CurrencyField(Field)` not suffice?

**Pull request closed without merging**